src/main/java/appeng/items/tools/ToolAdvancedNetworkTool.java

```diff
@@ -0,0 +1,108 @@
+import appeng.transformer.annotations.Integration.InterfaceList;
+import appeng.util.Platform;
+import buildcraft.api.tools.IToolWrench;
+import cofh.api.item.IToolHammer;
+
+@InterfaceList(
+        value = { @Interface(iface = "cofh.api.item.IToolHammer", iname = IntegrationType.CoFHWrench),
+                @Interface(iface = "buildcraft.api.tools.IToolWrench", iname = IntegrationType.BuildCraftCore) })
+public class ToolAdvancedNetworkTool extends AEBaseItem implements IGuiItem, IAEWrench, IToolWrench, IToolHammer {
+
+    public ToolAdvancedNetworkTool() {
+        super(Optional.absent());
+
+        this.setFeature(EnumSet.of(AEFeature.AdvancedNetworkTool));
+        this.setMaxStackSize(1);
+        this.setHarvestLevel("wrench", 0);
+    }
+
+    @Override
+    public ItemStack onItemRightClick(final ItemStack it, final World w, final EntityPlayer p) {
+        if (Platform.isClient()) {
+            final MovingObjectPosition mop = ClientHelper.proxy.getMOP();
+
+            if (mop == null) {
+                this.onItemUseFirst(it, p, w, 0, 0, 0, -1, 0, 0, 0);
+            } else {
+                final int i = mop.blockX;
+                final int j = mop.blockY;
+                final int k = mop.blockZ;
+
+                if (w.getBlock(i, j, k).isAir(w, i, j, k)) {
+                    this.onItemUseFirst(it, p, w, 0, 0, 0, -1, 0, 0, 0);
+                }
+            }
+        }
+
+        return it;
+    }
+
+    @Override
+    public boolean onItemUseFirst(final ItemStack is, final EntityPlayer player, final World world, final int x,
+            final int y, final int z, final int side, final float hitX, final float hitY, final float hitZ) {
+        // TODO
+
+        return true;
+    }
+
+    @Override
+    public boolean doesSneakBypassUse(final World world, final int x, final int y, final int z,
+            final EntityPlayer player) {
+        return true;
+    }
+
+    @Override
+    public boolean canWrench(final ItemStack is, final EntityPlayer player, final int x, final int y, final int z) {
+        return true;
+    }
+
+    @Override
+    public boolean canWrench(final EntityPlayer player, final int x, final int y, final int z) {
+        return true;
+    }
+
+    @Override
+    public void wrenchUsed(final EntityPlayer player, final int x, final int y, final int z) {
+        player.swingItem();
+    }
+
+    @Override
+    public boolean isUsable(ItemStack itemStack, EntityLivingBase entityLivingBase, int x, int y, int z) {
+        return true;
+    }
+
+    @Override
+    public void toolUsed(ItemStack itemStack, EntityLivingBase entity, int x, int y, int z) {
+        entity.swingItem();
+    }
+
+    @Override
+    public IGuiItemObject getGuiObject(final ItemStack is, final World world, final int x, final int y, final int z) {
+        final TileEntity te = world.getTileEntity(x, y, z);
+        return new NetworkToolViewer(is, (IGridHost) (te instanceof IGridHost ? te : null));
```

Comment:
```suggestion
        return new GuiAdvancedNetworkTool(is, (IGridHost) (te instanceof IGridHost ? te : null));
```
It is better to save common part for all related entities

Reply:
I just copy this method from `ToolNetworkTool.java`, I will replace it when I figure out how these method works.